what4/src/What4/Protocol/SMTLib2.hs

```diff
@@ -1151,26 +1152,28 @@ shutdownSolver _solver p = do
 -----------------------------------------------------------------
 -- Checking solver version bounds
 
-mkChunks :: Word -> [Word] -> NonEmpty Versions.VChunk
-mkChunks maj vs = fmap ( (:| []) . Versions.Digits) (maj :| vs)
-
 -- | The minimum (inclusive) version bound for a given solver.
 --
 -- The keys come from @'smtWriterName'@ in @'WriterConn'@.
 -- See also https://github.com/GaloisInc/crucible/issues/194
 solverMinVersions :: Map String Version
 solverMinVersions =
-  [ -- TODO: Why is this verion required?
-    ( "Yices"
-    , Version { _vEpoch = Nothing, _vChunks = mkChunks 2 [6, 1], _vRel = [], _vMeta = [] }
-    )
+  [ ( "Yices", $(ver "2.6.1") )
+  , ( "Z3"   , $(ver "4.8.7") )
+  , ( "CVC4" , $(ver "1.7") )
+  , ( "STP"  , $(ver "3.2.1") )
   ]
 
 -- | The maximum (non-inclusive) version bound for a given solver.
 --
 -- The keys come from @'smtWriterName'@ in @'WriterConn'@.
 solverMaxVersions :: Map String Version
-solverMaxVersions = []
+solverMaxVersions =
+  [ ( "Yices", $(ver "2.7") )
+  , ( "Z3"   , $(ver "4.9") )
+  , ( "CVC4" , $(ver "1.9") )
+  , ( "STP"  , $(ver "3.3") )
+  ]
```

Comment:
I'm wondering if we should maintain these min and max table values in an external file that's read in to make it easier to find and/or change.  Then we can also put a link to that file in the README and just maintain the information in a single place.

That would force both of these to be in the IO monad, but currently they are only called from an IO monad location.

Reply:
Well, we can do the whole thing in TH, so reading from a file would be no problem, the type of the function wouldn't need to change.

I agree it would be a good idea generally to maintain a canonical place for the info.

Reply:
Good point about TH: anyone that will be using this library should have no trouble reading these TH tables. If we put them in their own file for clarity and link to that file from the README I think that keeps all the information clear and locatable.

Is the test failure just the unneeded BV import?

Reply:
No, I think it's a linker error :-(

Reply:
```
<command line>: libz.so.1: cannot open shared object file: No such file or directory
```

Reply:
I wonder if this is related to the fact that GitHub recently switched over their `ubuntu-latest` runners from using GitHub 18.04 to 20.04. I'm currently debugging a similar issue [here](https://github.com/GaloisInc/saw-script/pull/1057#issuecomment-771916725).